### PR TITLE
[FLINK-33750][config] Remove unused deprecated option

### PR DESCRIPTION
**Pull request by Sxnan**
## What is the purpose of the change

Remove unused deprecated option. Note that this PR left some deprecated ConfigOption. 

- Options related to restart strategy, such as "hidden.execution.retries"
- Options related to legacy memory configuration, such as "jobmanager.heap.size"
- Options related to netty network shuffle
- Options related to serizliation, such as "pipeline.auto-type-registration"
- Options related to queryable state, QueryableStateOptions

## Brief change log

- Remove deprecated option in flink-core
- Remove deprecated option in flink-metrics-prometheus
- Remove deprecated option StreamPipelineOptions

## Verifying this change

This change is already covered by existing tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: yes
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no
 
## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/NettyShuffleEnvironmentConfiguration.java [hunk 1]
```diff
@@ -356,27 +354,9 @@ public static NettyShuffleEnvironmentConfiguration fromConfiguration(
...
-                        NettyShuffleEnvironmentOptions.SHUFFLE_COMPRESSION_CODEC.key());
-            }
-        }
+        boolean batchShuffleCompressionEnabled = compressionCodec != CompressionCodec.NONE;
 
-        int maxNumConnections =
-                Math.max(
-                        1,
-                        configuration.get(NettyShuffleEnvironmentOptions.MAX_NUM_TCP_CONNECTIONS));
+        int maxNumConnections = 1;
```
Comment (1996fanrui) [inline flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/NettyShuffleEnvironmentConfiguration.java:359]: Same with the last comment, if it's always 1, this parameter may be not needed, and some code can be simplified.

**Review by 1996fanrui — COMMENTED** [review]
Thanks @Sxnan for the contribution!

Overall LGTM, I left some minor comments, please take a look in your free time, thanks~

**Comment by Sxnan** [comment]
@1996fanrui Thanks for the review! I updated the PR. PTAL.

**Review by 1996fanrui — APPROVED** [review]
Thanks @Sxnan for the update!

LGTM

**Pull request merged**